torchgeo/datasets/ssl4eo_downstream_landsat.py

```diff
@@ -0,0 +1,224 @@
+        "oli_sr": "ssl4eo_l_oli_*",
+    }
+
+    year_dict = {
+        "tm_toa": 2011,
+        "etm_toa": 2019,
+        "etm_sr": 2019,
+        "oli_tirs_toa": 2019,
+        "oli_sr": 2019,
+    }
+
+    RGB_INDICES = {
+        "tm_toa": [2, 1, 0],
+        "etm_toa": [2, 1, 0],
+        "etm_sr": [2, 1, 0],
+        "oli_tirs_toa": [3, 2, 1],
+        "oli_sr": [3, 2, 1],
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        input_sensor: str = "etm_toa",
+        mask_product: str = "cdl",
+        split: str = "train",
+        transforms: Optional[Callable[[dict[str, Tensor]], dict[str, Tensor]]] = None,
+    ) -> None:
+        """Initialize a new SSL4EODownstream instance.
+
+        Args:
+            root: root directory where dataset can be found
+            input_sensor: one of ['etm_toa', 'etm_sr', 'oli_tirs_toa, 'oli_sr']
+            mask_product: mask target matched to input_sensor
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+
+        Raises:
+            AssertionError: if ``input_sensor`` argument is invalid
+            AssertionError: if ``mask_product`` argument is invalid
+            AssertionError: if ``split`` argument is invalid
+        """
+        assert (
+            input_sensor in self.valid_input_sensors
+        ), f"Only supports one of {self.valid_input_sensors}, but found {input_sensor}."
+        self.input_sensor = input_sensor
+        assert (
+            mask_product in self.valid_mask_products
+        ), f"Only supports one of {self.valid_mask_products}, but found {mask_product}."
+        self.mask_product = mask_product
+        assert (
+            split in self.valid_splits
+        ), f"Only supports one of {self.valid_splits}, but found {split}."
+        self.split = split
+
+        self.root = root
+        self.transforms = transforms
+
+        self.sample_collection = self.retrieve_sample_collection()
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            image and sample
+        """
+        img_path, mask_path = self.sample_collection[index]
+
+        sample = {
+            "image": self._load_image(img_path),
+            "mask": self._load_mask(mask_path),
+        }
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.sample_collection)
+
+    def retrieve_sample_collection(self) -> list[tuple[str]]:
+        """Retrieve paths to samples in data directory."""
+        img_data_dir = self.image_root.replace("*", self.input_sensor)
```

Comment:
If we're only using `*` as something to replace, why not remove it and concatenate instead of replace?